src/main/java/com/puppycrawl/tools/checkstyle/api/metadata/JavadocMetadataScraper.java

```diff
@@ -0,0 +1,128 @@
+package com.puppycrawl.tools.checkstyle.api.metadata;
+
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck;
+
+public class JavadocMetadataScraper extends AbstractJavadocCheck {
+    private static final Pattern TAGS_TO_IGNORE = Pattern.compile("([A-Z]+_*)+[A-Z]+");
+    private ModuleDetails moduleDetails;
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+                JavadocTokenTypes.PARAGRAPH,
+                JavadocTokenTypes.LI
```

Comment:
I still think you do not need any special token, you just need to execute on root node and slice and dice javadoc as required

Reply:
Yes, I tried that approach, but  I feel this will be a bit easier as it will reduce the slicing and dicing part. Because elements of type `<li>` are quite deep and its easier to get them through visited token approach. These will result in less code written.